alerts/UA00002.json

```diff
@@ -1,15 +1,15 @@
 {
   "dateRaised": 20201028,
   "affectedFirmware": ["all"],
-  "hardwareLimited": [],
-  "description": "When using SBUS output, if the RC receiver is lost the IOMCU will invert the SBUS output momentarily whilst searching for an FPort input. This can lead to a loss of signal on the SBUS output port. This only affects boards with an IOMCU",
-  "mitigation": "Do not use the SBUS output port",
+  "hardwareLimited": ["CUAVv5", "CubeBlack", "CubeBlack+", "CubeGreen-solo", "CubeOrange", "CubePurple", "CubeSolo", "CubeYellow", "DrotekP3Pro", "Durandal", "fmuv2", "fmuv3", "fmuv5", "mRoX21", "mRoX21-777", "Pix32v5", "Pixhawk1", "Pixhawk1-1M", "Pixhawk4"],
+  "description": "When using SBUS output, if the RC receiver is lost the IOMCU will invert the SBUS output momentarily whilst searching for an RC input. This can lead to a loss of signal on the SBUS output port. This only affects boards with an IOMCU that use the default SBUS Output port.",
+  "mitigation": "Do not use the default SBUS OUT port. Use an alternative UART for SBUS Output.",
```

Comment:
I really think the mitigation should be to upgrade to Copter-4.0.5 or Plane-4.0.6 (or later).

Reply:
The "mitigation" field is defined (from https://ardupilot.org/dev/docs/user-alerts-developer.html) as the pre-patch method to mitigate the issue.

Any GCS or website ingesting this data can, of course, extend the "mitigation" field to add in "upgrade to XXX", as is done in https://firmware.ardupilot.org/useralerts/alerts.html. Note UA00001 does not have "upgrade to..." in the userAlerts repo, but the website adds on the "upgrade to..." text automatically.

There are a few reasons we do this:
1) A user may not be in a position to upgrade right now, but can mitigate the issue in the meantime.
2) Due to the different release cycles of copter/plane/rover, firmwares may not be patched at the same time. For example an ArduRover user may need to wait N weeks for a patched version to upgrade to, whilst ArduCopter users can upgrade now.

Reply:
.. say we have a crash-o-matic bug though, like a watchdog reset that happens somewhat randomly.  We are not going to go back and re-publish a new 4.x.x release.  Instead we will release a new version 4.x.x+1.  The mitigation will be to upgrade to the later version.

So if we can have situations where upgrading to the later version is the mitigation then I think this issue is one of those cases.  I think is we don't support a mitigation which is to upgrade then we should add it.

Reply:
>The mitigation will be to upgrade to the later version.

That's what the versionFrom and versionFixed fields are for. The idea is that any GCS (or user) using the User Alerts can compare these fields to the connected vehicle's version and decide whether the alert is applicable.

> So if we can have situations where upgrading to the later version is the mitigation then I think this issue is one of those cases

From my understanding, there are 2 mitigations available: 1) Change the SBUS Out port via the SERIALX_PROTOCOL param 2) Upgrade to Copter 4.0.5 / Plane 4.0.7

1) goes in the mitigation field and 2) goes in the versionFixed field. This is what we've done previously (see https://github.com/ArduPilot/useralerts/blob/master/alerts/UA00001.json)

Reply:
I think the best mitigation is to upgrade.  Why wouldn't we put the best upgrade in the mitigation field instead of the 2nd best mitigation?